src/core/bsearch.rkt

```diff
@@ -100,6 +101,20 @@
     (cons val (random-ref pts)))
   (apply mk-pcontext (cdr (batch-prepare-points evaluator new-sampler))))
 
+(define cache (make-hash))
```

Comment:
I would use `define-resetter`; search the codebase for examples. Basically that's just a way of making a hash table, but clearing it between benchmarks (so memory doesn't grow forever)